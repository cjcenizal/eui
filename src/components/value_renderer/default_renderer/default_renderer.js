import { isArray, isBoolean, isDate, isNaN, isNil, isNumber, isString } from 'lodash';
import {
  formatBoolean,
  formatDate,
  formatNumber,
  formatText,
} from '../../../services';
import { join } from '../compound';

export const defaultRenderer = (value) => {
  if (isNil(value) || isNaN(value)) {
    return '';
  }
  if (isString(value)) {
    return formatText(value);
  }
  if (isDate(value)) {
    return formatDate(value);
  }
  if (isBoolean(value)) {
    return formatBoolean(value);
  }
  if (isNumber(value)) {
    return formatNumber(value);
  }
  if (isArray(value)) {
    return join(defaultRenderer)(value);
  }
  // TODO not sure if we want that.. the (+) is that we show something, the (-) is that it's very technical
  return JSON.stringify(value);
};
