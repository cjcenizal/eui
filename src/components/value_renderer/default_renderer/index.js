export { defaultRenderer } from './default_renderer';
