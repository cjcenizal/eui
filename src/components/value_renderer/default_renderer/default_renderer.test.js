import { defaultRenderer } from './default_renderer';

describe('Value Renderer', () => {
  describe('defaultRenderer', () => {

    test('boolean value', () => {
      expect(defaultRenderer(true)).toMatchSnapshot();
      expect(defaultRenderer(false)).toMatchSnapshot();
    });

    test('numeric value', () => {
      expect(defaultRenderer(1234.567)).toMatchSnapshot();
    });

    test('string value', () => {
      expect(defaultRenderer('value')).toMatchSnapshot();
    });

    test('date value', () => {
      const value = new Date(1999, 0, 1, 2, 3, 4, 5);
      expect(defaultRenderer(value)).toMatchSnapshot();
    });

    test('array of dates', () => {
      const dates = [ new Date(1999, 0, 1, 2, 3, 4, 5) ];
      expect(defaultRenderer(dates)).toMatchSnapshot();
    });

    test('object value', () => {
      const obj = { key: 'value' };
      expect(defaultRenderer(obj)).toMatchSnapshot();
    });

  });
});
