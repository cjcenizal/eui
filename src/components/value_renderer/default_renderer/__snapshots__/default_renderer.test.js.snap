// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value Renderer defaultRenderer array of dates 1`] = `"1 Jan 1999 02:03"`;

exports[`Value Renderer defaultRenderer boolean value 1`] = `"Yes"`;

exports[`Value Renderer defaultRenderer boolean value 2`] = `"No"`;

exports[`Value Renderer defaultRenderer date value 1`] = `"1 Jan 1999 02:03"`;

exports[`Value Renderer defaultRenderer numeric value 1`] = `"1234.567"`;

exports[`Value Renderer defaultRenderer object value 1`] = `"{\\"key\\":\\"value\\"}"`;

exports[`Value Renderer defaultRenderer string value 1`] = `"value"`;
